### PR TITLE
[make_watertight] regression tests

### PR DESCRIPTION
**Pull request by Pshriwise**
This PR adds new test classes for make_watertight regression tests which will only run with DAGMC's nightly builds (so they won't be run as part of this PR). 

The test models are served from UW box and retrieved by travis via `wget` in the docker instance.

### CODE REVIEW

File: .run_tests.sh [hunk 1]
```diff
@@ -26,6 +26,12 @@ cd tests
 ./make_watertight_no_curve_sphere_tests
 ./make_watertight_cylinder_tests
 ./make_watertight_cone_tests
+# if this is not a pull request, run regression tests
+if [ ! TRAVIS_PULL_REQUEST ] ; then
+wget https://uwmadison.box.com/shared/static/jns2dfu47euucwexe74l4qroxneelp9u.gz -O mw_reg_test_files.tar.gz -o wget.out
```
Comment (makeclean) [inline .run_tests.sh]: indent with tabs

Comment (makeclean) [inline .run_tests.sh]: I thought we were going to obfusciate this link and keep in a Travis encrypted variable?

Comment (Pshriwise) [inline .run_tests.sh]: That's correct. Forgot in my excitement that these are going to be part of the nightly's now. Updated that link to be a Travis environment variable.

**Comment by Pshriwise** [comment]
I've now added necessary environment variables to the docker instance needed to make sure the tests only run for non-pull requests and that the regression test file location can be found by wget using the encrypted travis variable. 

*Note:* any travis account we rely upon will need these variables in the future. So if people begin to test on dagmc forks using travis, they'll need these environment variables as well.

Here is also some proof that the regression tests are successful for a branch push on my fork. 
reg_tests

**Comment by makeclean** [comment]
Looks great :)

**Pull request merged**